code/dynamic_programming/Array_Median/median.c

```diff
@@ -0,0 +1,53 @@
+#include <stdio.h>
+
+void swap(float *p,float *q) {
+   float t;
+   
+   t=*p; 
+   *p=*q; 
+   *q=t;
+}
+
+void sort(float a[],float n) { 
+   int i,j;
+
+   for(i=0;i<n-1;i++) {
+      for(j=0;j<n-i-1;j++) {
+         if(a[j]>a[j+1])
+            swap(&a[j],&a[j+1]);
+      }
+   }
+}
+
+int main() {
+	int i;
+	int n;
+	printf("please enter the size\n");
+	scanf("%d",&n);
+	float a[n];
```

Comment:
While Variable length arrays (VLAs) are **mostly** supported in C99, and definitely after it, some compilers still could define the macro ` _STDC_NO_VLA_` ([Source](https://stackoverflow.com/questions/16672322/malloced-array-vs-variable-length-array#comment23988958_16672322)) and not implement variable length arrays. So, just to be safe, you should use `malloc`, as follows

```c
float* a = (float*)malloc(n * sizeof(float));

// ... code for quicksort ...

free(a);
```
This will guarantee standards compliance for your code, regardless of compiler.